FGEGraphics/ClientSystem/ShaderLocations.cs

```diff
@@ -42,4 +46,309 @@ public static class Common2D
+        /// Uniform: eye_position
+        /// </summary>
+        public const int EYE_POSITION = 20;
+
+        /// <summary>
+        /// What color to desaturate too. Default is an orange-ish color.
+        /// Uniform: desaturationColor
+        /// </summary>
+        public const int DESATURATION_COLOR = 21;
+
+        /// <summary>
+        /// The full 3D projection matrix.
+        /// Uniform: proj_mat
+        /// </summary>
+        public const int PROJECTION_MATRIX = 22;
+
+        /// <summary>
+        /// How wide the screen is.
+        /// Uniform: width
+        /// </summary>
+        public const int SCREEN_WIDTH = 24;
+
+        /// <summary>
+        /// How tall the screen is.
+        /// Uniform: height
+        /// </summary>
+        public const int SCREEN_HEIGHT = 25;
+
+        /// <summary>
+        /// A timer value, in seconds. Simply used for things that move.
+        /// Uniform: time
+        /// </summary>
+        public const int TIME = 26;
+
+        /// <summary>
+        /// How much motion blur to apply, and in what direction.
+        /// Uniform: mot_blur
+        /// </summary>
+        public const int MOTION_BLUR = 27;
+
+        /// <summary>
+        /// Whether to gray-scale the view.
+        /// Uniform: do_grayscale
+        /// </summary>
+        public const int DO_GRAYSCALE = 28;
+    }
+
+    /// <summary>Locations used in the transponly shader.</summary>
+    public static class TranspOnly
+    {
+        /// <summary>
+        /// Amount of desaturation to apply.
+        /// Uniform: desaturationAmount
+        /// </summary>
+        public const int DESATURATION = 4;
+
+        /// <summary>
+        /// Screen size.
+        /// Uniform: u_screen_size
+        /// </summary>
+        public const int SCREEN_SIZE = 8;
+
+        /// <summary>
+        /// Helper for lights used.
+        /// Uniform: lights_used_helper
+        /// </summary>
+        public const int LIGHTS_USED_HELPER = 9;
+
+        /// <summary>
+        /// The matrices of the light sources (essentially their point of view this is used for shadow mapping).
+        /// Uniform: shadow_matrix_array
+        /// </summary>
+        public const int SHADOW_MATRIX_ARRAY = 20;
+
+        /// <summary>
+        /// The distance fog should be around.
+        /// Uniform: fogDist
+        /// </summary>
+        public const int FOG_DISTANCE = 13;
+
+        /// <summary>
+        /// Camera position, relative to rendering origin.
+        /// Uniform: cameraPos
+        /// </summary>
+        public const int CAMERA_RELATIVE_POSITION = 14;
+    }
+
+    /// <summary>Locations used in the transpadder shader.</summary>
+    public static class TranspAdder
+    {
+        /// <summary>
+        /// The amount of lights used.
+        /// Uniform: lcount
+        /// </summary>
+        public const int LIGHTS_USED = 3;
+    }
+
+    /// <summary>Locations used in the clearer shader.</summary>
```

Comment:
This... can probably just not be added here. It's a very special case / experiment thing that maybe probably shouldn't exist anyway